### PR TITLE
Modify kafka input to include parsing multiple messages

### PR DESCRIPTION
**Pull request by narph**
If the fileset using this input expects to receive multiple messages bundled under a specific field then the config option ExpandEventListFromField value can be assigned the name of the json group object.
For example in the case of azure fielsets the events are found under the json object "records".
```
 {
"records": [ {event1}, {event2}]
}
```

Initial draft PR https://github.com/elastic/beats/pull/13776

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -333,6 +333,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add input httpjson. {issue}13545[13545] {pull}13546[13546]
 - Filebeat Netflow input: Remove beta label. {pull}13858[13858]
 - Remove `event.timezone` from events that don't need it in some modules that support log formats with and without timezones. {pull}13918[13918]
+- Add YieldEventsFromField config option in the kafka input. {pull}13965[13965]
```
Comment (faec) [inline CHANGELOG.next.asciidoc]: Non-blocking preference: I think `ExpandEventsFromField` or even `ExpandEventListFromField` might be clearer. Either way, please document the new field in `filebeat/docs/inputs/input-kafka.asciidoc`

I like the greater flexibility of this less Azure-specific approach :-)

File: filebeat/input/kafka/input.go [hunk 2]
```diff
@@ -307,8 +325,29 @@ func (h *groupHandler) ack(message *sarama.ConsumerMessage) {
...
+		events := h.createEvents(sess, claim, msg)
+		for _, event := range events {
+			h.outlet.OnEvent(event)
+		}
 	}
 	return nil
 }
+
+// parseMultipleMessages will try to split the message into multiple ones based on the group field provided by the configuration
+func (h *groupHandler) parseMultipleMessages(bMessage []byte) []string {
```
Comment (faec) [inline filebeat/input/kafka/input.go]: There should be at least one test exercising the multiple-message code path. Kafka is awkward to unit test but you can probably add something to `kafka_integration_test.go` without too much trouble.

File: filebeat/input/kafka/config.go [hunk 3]
```diff
@@ -34,22 +34,23 @@ import (
...
+	ConsumeBackoff           time.Duration     `config:"consume_backoff" validate:"min=0"`
+	WaitClose                time.Duration     `config:"wait_close" validate:"min=0"`
+	MaxWaitTime              time.Duration     `config:"max_wait_time"`
+	IsolationLevel           isolationLevel    `config:"isolation_level"`
+	Fetch                    kafkaFetch        `config:"fetch"`
+	Rebalance                kafkaRebalance    `config:"rebalance"`
+	TLS                      *tlscommon.Config `config:"ssl"`
+	Username                 string            `config:"username"`
+	Password                 string            `config:"password"`
+	ExpandEventListFromField string            `config:"expand_events_from_field"`
```
Comment (faec) [inline filebeat/input/kafka/config.go]: Either name is fine but the struct name should match the config name :-)

**Comment by narph** [comment]
jenkins test this

File: filebeat/docs/inputs/input-kafka.asciidoc [hunk 4]
```diff
@@ -127,6 +127,17 @@ Kafka fetch settings:
 *`max`*:: The maximum number of bytes to read per request. Defaults to 0
 (no limit).
 
+===== `expand_event_list_from_field`
+
+If the fileset using this input expects to receive multiple messages bundled under a specific field then the config option `expand_event_list_from_field` value can be assigned the name of the json group object.
```
Comment (faec) [inline filebeat/docs/inputs/input-kafka.asciidoc]: "the name of the json group object" -> "the name of the field" is I think clearer about what value should go in the config

"fielsets" -> "filesets" in the following line

**Pull request merged**